src/cache.rs

```diff
@@ -0,0 +1,300 @@
+//! This module provides caching utils
```

Comment:
Here's some inspiration to solve your imminent issue of compiling on WASM in a transparent manner:
https://github.com/rust-lang/rust/blob/master/compiler/rustc_data_structures/src/memmap.rs